.github/workflows/test.yml

```diff
@@ -179,6 +185,26 @@ jobs:
           # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
           # skip-build-cache: true
 
+  go_unit:
+    name: Go Unit Tests
+    needs: determine_jobs
+    if: needs.determine_jobs.outputs.go == 'true'
```

Comment:
Calling out that this will get run a lot more than in currently does. ATM we run on `cli/**` changes, but `go` is triggered whenever `shim/**`, `packages/**` are changed. I don't think any of those can affect the unit tests.

Reply:
I can change that. I copied it from Go Lint, but that one could be scoped down as well